zerver/migrations/0433_create_nobody_system_group.py

```diff
@@ -0,0 +1,50 @@
+# Generated by Django 4.1.7 on 2023-03-27 03:00
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def create_nobody_system_user_group_for_existing_realms(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    Realm = apps.get_model("zerver", "Realm")
+    UserGroup = apps.get_model("zerver", "UserGroup")
+    NOBODY_GROUP_NAME = "@role:nobody"
+    NOBODY_GROUP_DESCRIPTION = "Nobody in this organization"
+
+    groups_to_create = []
+    for realm in Realm.objects.all():
+        groups_to_create.append(
+            UserGroup(
+                name=NOBODY_GROUP_NAME,
+                description=NOBODY_GROUP_DESCRIPTION,
+                realm=realm,
+                is_system_group=True,
+            )
+        )
+
+    UserGroup.objects.bulk_create(groups_to_create)
```

Comment:
If it's easy, it would probably be a good idea to batch this operation; I'm convinced creating this many objects in a single transaction is a good idea.

Reply:
I am confused by the comment above. Do we want to batch this or not?

We can do batching using Realm ID. But I think we have done similar operations in a single transaction previously which inlcudes all realms.

Reply:
If previous user group migrations haven't required a transaction, maybe it's fine; I guess one thing that helps here is that likely nothing will fail to work while this migration is running. We can try it as written and adjust it later if needed.

Reply:
Just to be clear, in the 0382 migration which added system groups, the `UserGroup`, `GroupGroupMembership` and `UserGroupMembership` objects for each realm are created in an atomic transaction.

In the above comment, I was talking about migrations like one which added realm creation `RealmAuditLog` objects for all realms. These are similar in sense that a single new obejct is created for all realms.